pkg/k8s/annotations.go

```diff
@@ -0,0 +1,58 @@
+package k8s
+
+import (
+	"context"
+
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// AnnotateObject adds the specified annotations to the object
+func AnnotateObject(ctx context.Context, c client.Client, obj client.Object, annotations map[string]string) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		_obj := obj.DeepCopyObject().(client.Object)
+
+		err := c.Get(ctx, client.ObjectKeyFromObject(obj), _obj)
+		if err != nil {
+			return err
+		}
+
+		orig := _obj.DeepCopyObject().(client.Object)
+
+		a := _obj.GetAnnotations()
+		if a == nil {
+			a = make(map[string]string)
+		}
+
+		for k, v := range annotations {
+			a[k] = v
+		}
+		_obj.SetAnnotations(a)
+
+		return c.Patch(ctx, _obj, client.MergeFrom(orig))
```

Comment:
There is no need to use `RetryOnConflict` when using `Patch`. `Patch` doesn't return conflict error: https://kubernetes.io/docs/reference/using-api/api-concepts/#patch-and-apply

I suggest using `Update` here

Reply:
if `Patch` doesn't return conflict error, then I'll remove `retry.OnConflict`

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1487/commits/50fbe4ea337b258b3238623e1b722f8916498ea8